src/components/Table.tsx

```diff
@@ -76,7 +76,8 @@ export type ColumnDef<TableRowData extends BaseTableRowData | CustomRowConfig> =
   tag?: React.ReactNode;
   colspan?: number;
   childColumns?: ColumnDef<TableRowData>[];
-  getCellValue: (row: TableRowData) => string | number;
+  // if sorting is false, can pass undefined for getCellValue
+  getCellValue: ((row: TableRowData) => string | number | undefined | null) | undefined;
```

Comment:
hmm to enforce this in typescript would it be possible to do something like 

```
type SortableColumnDef = {
  allowsSorting: true;
  getCellValue: ((row: TableRowData) => string | number | undefined | null);
}

type NonSortableColumnDef = {
  allowsSorting: false;
}

type ColumnDef<...>  = {
...
} & (SortableColumnDef || NonSortableColumnDef)
```

alternatively, is this just equivalent to `getCellValue?: ((row: TableRowData) => string | number | undefined | null)`

Reply:
another thought, is it necessary for us to capture `undefined` and `null` in the return types? (I sort of prefer just using one or the other, `null` seems to make more sense in this case)

Reply:
YEES great idea! That's much cleaner.